### DISCUSSION TITLE
Nested dynamic branches

### DISCUSSION THREAD

**Discussion by stenw** [ropensci/targets#971] [Help]
What would be the best way to create nested dynamic branches (not crossed)?
With static branching it could be done like in the silly example below but the simplest way to do it with dynamic branching seems to let each branch retain the infor of its parents or to create a nested tibble bot of which seem cumbersome as the objects to work with become larger.

---

library(targets)
library(tarchetypes)
library(dplyr)

outer_names <- list(A=1:2) # not actually relevant in this silly example
inner_names <- list(a=1:2)

random_matrix <- function(min, max){
  n <-  floor(runif(1, min, max+1))
  m <-  floor(runif(1, min, max+1))
  v <- rnorm(n*m)
  matrix(v,nrow =  n,ncol =  m)
}

random_entry <- function(M){
  n <- NROW(M)
  m <- NCOL(M)
  i <-  floor(runif(1, 1, n+1))
  j <-  floor(runif(1, 1, m+1))

  M[i, j]
}

list(
  tar_map(values=outer_names,
   tar_target(
    mat,
    random_matrix(3, 5)
   ),
   tar_map(values=inner_names,
    tar_target(entry,
              print(random_entry(mat))
    )
   )
  )
)

**Comment by wlandau** [comment]
I'm not sure I understand what you are going for, but I do recommend looking at batching: splitting tasks into targets with dynamic branching, then looping over things within each branch target. It works well with data frames that you can row-bind together. I recommend checking out `tarchetypes::tar_rep()` and `tarchetypes::tar_map_rep()` for the case when you want to perform a bunch of repeated simulations.
